etc/completions

```diff
@@ -1,5 +1,807 @@
-$command $opt* ($revision|$path|$remote)* --? $path+
+(add|stage) $opt* --? $path+
+  --all
+  --dry-run
+  --edit
+  --force
+  --ignore-errors
+  --ignore-missing
+  --ignore-removal
+  --intent-to-add
+  --interactive
+  --no-all
+  --no-ignore-removal
+  --refresh
+  --update
+  --verbose
+
+blame $opt* $revision? --? $path
+  --incremental
+  --line-porcelain
+  --no-progress
+  --porcelain
+  --progress
+  --root
+  --score-debug
+  --show-email
+  --show-name
+  --show-number
+  --show-stats
+  --abbrev $anything
+  --contents $path
+  --date $anything
+  --encoding $anything
+  --reverse $revision
+  --since $anything
+  -S $path
+
+# Listing
```

Comment:
Nope, it's a custom file format described in this manual page: https://github.com/thoughtbot/gitsh/blob/1f7c9615384bf7661cb5557918350c46ebc99b46/man/man5/gitsh_completions.5.in

It's a bit like regular expressions, but the atoms are command arguments not characters.